scripts/check_compliance.py

```diff
@@ -895,211 +891,225 @@ def init_logs(cli_arg):
                  logging.getLevelName(logger.getEffectiveLevel()))
 
 
+def set_pending():
+    # Sets 'pending' status for all tests for the commit given by --sha
 
-def set_status(repo, sha):
-    """
-    Set status on Github
-    :param repo:  repoistory name
-    :param sha:  pull request HEAD SHA
-    :return:
-    """
-
-    if 'GH_TOKEN' not in os.environ:
-        return
-    github_token = os.environ['GH_TOKEN']
-    github_conn = Github(github_token)
-
-    repo = github_conn.get_repo(repo)
-    commit = repo.get_commit(sha)
     for testcase in ComplianceTest.__subclasses__():
-        test = testcase(None, "")
-        print("Creating status for %s" % (test._name))
-        commit.create_status('pending',
-                             '%s' % test._doc,
-                             'Checks in progress',
-                             '{}'.format(test._name))
+        print("Creating pending status for " + testcase._name)
+        github_commit.create_status(
+            'pending', testcase._doc, 'Checks in progress', testcase._name)
 
 
-def report_to_github(repo, pull_request, sha, suite, docs):
-    """
-    Report test results to Github
-
-    :param repo: repo name
-    :param pull_request:  pull request number
-    :param sha:  pull request SHA
-    :param suite:  Test suite
-    :param docs:  documentation of statuses
-    :return: nothing
-    """
-
-    if 'GH_TOKEN' not in os.environ:
-        return
-
-    username = os.environ.get('GH_USERNAME', 'zephyrbot')
+def report_test_results_to_github(suite):
+    # Reports test results to Github.
+    #
+    #   suite: Test suite
 
-    github_token = os.environ['GH_TOKEN']
-    github_conn = Github(github_token)
+    fail_msg = FAIL_MSG + "\n\n"
+    n_failures = 0
 
-    repo = github_conn.get_repo(repo)
-    gh_pr = repo.get_pull(pull_request)
-    commit = repo.get_commit(sha)
+    print("reporting results to GitHub")
 
-    comment = "Found the following issues, please fix and resubmit:\n\n"
-    comment_count = 0
+    name2doc = {testcase._name: testcase._doc
+                for testcase in ComplianceTest.__subclasses__()}
 
-    print("Processing results...")
+    def set_commit_status(status, msg):
+        # 'case' gets set in the loop.
+        # pylint: disable=undefined-loop-variable
+        github_commit.create_status(status, name2doc[case.name], msg,
+                                    case.name)
```

Comment:
I'm curious whether this requires special privileges or can any user (accidentally or not) override CI status. Good for testing this script, maybe not as good for creating github confusion :-)
Only for their own commits maybe?